packages/flutter/test/material/autocomplete_test.dart

```diff
@@ -254,50 +285,131 @@ void main() {
+    )));
+
+    final double resultingHeight = await getOptionsHeight(tester, '');
+    expect(resultingHeight, equals(defaultHeight));
+  });
+
+  testWidgets('the options height restricts to max desired height',
+      (WidgetTester tester) async {
+    const double desiredHeight = 150.0;
+    await tester.pumpWidget(MaterialApp(
         home: Scaffold(
-          body: Autocomplete<String>(
-            initialValue: const TextEditingValue(text: 'lem'),
-            onSelected: (String selection) {
-              lastSelection = selection;
-            },
-            optionsBuilder: (TextEditingValue textEditingValue) {
-              return kOptions.where((String option) {
-                return option.contains(textEditingValue.text.toLowerCase());
-              });
-            },
+      body: Autocomplete<String>(
+        optionsBuilder: (TextEditingValue textEditingValue) {
+          return kOptions.where((String option) {
+            return option.contains(textEditingValue.text.toLowerCase());
+          });
+        },
+        maxOptionsHeight: desiredHeight,
+      ),
+    )));
+
+    /// entering "a" returns 9 items from kOptions so basically the
+    /// height of 9 options would be beyond `desiredHeight=150`,
+    /// so height gets restricted to desiredHeight.
+    final double resultingHeight = await getOptionsHeight(tester, 'a');
+
+    /// expected desired Height =150.0
+    expect(resultingHeight, equals(desiredHeight));
+  });
+
+  testWidgets(
+      'The height of options shrinks to height of resulting items, if less than maxHeight',
+      (WidgetTester tester) async {
+    const double maxOptionsHeight = 250.0;
+    await tester.pumpWidget(MaterialApp(
+        home: Scaffold(
+      body: Autocomplete<String>(
+        optionsBuilder: (TextEditingValue textEditingValue) {
+          return kOptions.where((String option) {
+            return option.contains(textEditingValue.text.toLowerCase());
+          });
+        },
+        maxOptionsHeight: maxOptionsHeight,
+      ),
+    )));
+
+    final Finder listFinder = find.byType(ListView);
+    expect(listFinder, findsNothing);
+
+    /// entering `a` returns 9 items(height > `maxOptionsheight`) from the kOptions
+    /// so height gets restricted to `maxOptionsheight =250`
+    final double nineItemsHeight = await getOptionsHeight(tester, 'a');
+    expect(nineItemsHeight, equals(maxOptionsHeight));
+
+    /// returns 2 Items (height < `maxOptionsHeight`)
+    /// so options height shrinks to 2 Items combined height
+    final double twoItemsHeight = await getOptionsHeight(tester, 'el');
+    expect(twoItemsHeight, lessThan(maxOptionsHeight));
+
+    /// returns 1 item (height < `maxOptionsHeight`) from `kOptions`
+    /// so options height shrinks to 1 items height
+    final double oneItemsHeight = await getOptionsHeight(tester, 'elep');
+    expect(oneItemsHeight, lessThan(twoItemsHeight));
+    testWidgets('initialValue sets initial text field value',
```

Comment:
This test is not formatted correctly, it looks like you're starting a new test within another test (started on line 329).

Reply:
thanks seems like that nesting of tests happened during resolving the merge conflict